packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -239,6 +254,19 @@ export class BlobManager {
     }
 
     public async getBlob(blobId: string): Promise<ArrayBufferLike> {
+        const request = { url: blobId };
+        if (this.tombstonedBlobs.has(blobId) ) {
+            // Note: if a user writes a request to look like it's viaHandle, we will also send this telemetry event
+            this.mc.logger.sendErrorEvent({
+                eventName: "GC_Tombstone_Blob_Requested",
+                url: request.url,
+                viaHandle: true,
+            });
+            if (this.throwOnTombstoneUsage) {
+                throw responseToException(createResponseError(404, "Blob removed by gc", request), request);
```

Comment:
I think you can directly throw an error here. Not sure if 404 is needed since blobs are requested via handles only.

Reply:
When we resolve handles we return a 200 when we correctly get the blob. I think we should stick to the 404 when we fail to retrieve the blob